crates/router/src/core/payments/transformers.rs

```diff
@@ -3036,6 +3036,13 @@ impl<F: Clone> TryFrom<PaymentAdditionalData<'_, F>> for types::SetupMandateRequ
                     .map(|customer| customer.clone().into_inner())
             });
         let amount = payment_data.payment_attempt.get_total_amount();
+
+        let webhook_url = Some(helpers::create_webhook_url(
+            router_base_url,
+            &attempt.merchant_id,
+            connector_name,
+        ));
```

Comment:
@Narayanbhat166 @sai-harsha-vardhan  Should we consider passing `MerchantConnectorAccountId` here, instead of the connector name? Basically update the `create_webhook_url()` function as well?

Reply:
Yes, it's always recommended to use `MerchantConnectorAccountId` in webhook url

Reply:
@sai-harsha-vardhan @SanchithHegde  I made a new fn `create_webhook_url_using_mca_id` ,but it can be only be used [here](https://github.com/juspay/hyperswitch/pull/6631/files#diff-6f19e8956b6645147dedca51bdadb551b2cfbe7ce0a7092e514a4dfa41deb3bcR226) . In the other places where `create_webhook_url()`  is currently used the MerchantConnectorAccountId value is `Optional`

Reply:
we will always have the `merchant_connector_id` when constructing the router data. So if it is not found you can raise an internal server error. 

I do not see any harm in updating the existing webhook function.

Reply:
updated the existing create_webhook_url fn